lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -278,32 +294,42 @@ def _on_update_status(self, event):
 class OpenSearchPluginError(OpenSearchError):
     """Exception thrown when an opensearch plugin is invalid."""
 
-
-class OpenSearchPluginRelationClusterNotReadyError(OpenSearchPluginError):
-    """Exception thrown when making API calls and cluster is not yet ready.
-
-    This exception in specific should be handled by classes managing the relations of plugins.
-    """
+    def __init__(self, msg):
+        super().__init__(msg)
 
 
 class OpenSearchPluginMissingDepsError(OpenSearchPluginError):
     """Exception thrown when an opensearch plugin misses installed dependencies."""
 
+    def __init__(self, name, deps):
+        super().__init__(f"Failed to install plugin: {name} - missing dependencies {deps}")
+        self.deps = deps
+
 
 class OpenSearchPluginInstallError(OpenSearchPluginError):
     """Exception thrown when opensearch plugin installation fails."""
 
+    def __init__(self, name):
+        super().__init__("Failed to install plugin: {}".format(name))
```

Comment:
This is fine I suppose, though redundant. 
It is already implied by the type of the exception itself on the stacktrace that a plugin failed to install. 
So: `raise OpenSearchPluginInstallError(f"Plugin: {plugin.name}")` means the same thing as `raise OpenSearchPluginInstallError(f"Failed to install plugin: {plugin.name}")`